core/src/main/kotlin/org/coepi/api/v4/reports/TCNReportService.kt

```diff
@@ -0,0 +1,55 @@
+package org.coepi.api.v4.reports
+
+import java.time.Clock
+import java.time.LocalDate
+import org.coepi.api.common.time.toUtcLocalDate
+import org.coepi.api.v4.dao.TCNReportRecord
+import org.coepi.api.v4.dao.TCNReportsDao
+import org.coepi.api.v4.toInterval
+
+interface TCNReportService {
+    /**
+     * Return a list of TCN reports. If [maybeDate] is null, the current UTC date will be used. If
+     * [maybeIntervalNumber] is null, the current interval will be used.
+     */
+    fun getReports(maybeDate: LocalDate?, maybeIntervalNumber: Long?): List<TCNReportRecord>
+
+    fun getReports(date: LocalDate, intervalNumber: Long): List<TCNReportRecord>
+
+    fun saveReport(reportData: ByteArray): TCNReportRecord
+}
```

Comment:
I'm not sure what this layer of abstraction is buying us ? Do we expect any other implementation of the service ? If we need to implement a new version, we probably won't follow the same request params.

Reply:
In my experience, interfaces serve as good documentation. Even if there's only ever one implementation, they make it easier for developers to see exactly what that implementation is _for_ without getting distracted by the details of how it's done.

Reply:
But good documentation can just be added to the method signature right ?  I get there are some folks can't help but get into the implementation rabbit hole, but I don't see that alone being a strong reason for adding an extra layer of abstraction . Else, we'll end up with and interface for every class we write.

Reply:
Adding interface implies having more then one implementation - it should be used for DAO-type objects where you want to keep the details of underlying storage implementation detail, and not for services.

Reply:
I don't agree that an interface implies more than one implementation. It's just a way of separating your specification of what the thing is meant to do from the actual implementation of the thing. They don't cost anything and they're helpful, so I tend to use them liberally.

But I don't think this needs to hold up the PR. 🙂 If this team prefers to only use interfaces when there are known multiple implementations, I can work with that. I'll update the PR accordingly. Happy to continue this conversation in other channels if people are interested.